sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -288,6 +288,9 @@ case class Join(
     condition: Option[Expression])
   extends BinaryNode with PredicateHelper {
 
+  /** Whether this join node is ordered before. Set true after it has been ordered. */
+  var ordered: Boolean = false
```

Comment:
But it's a `transform`, i.e. after we reroder a, b, c, d to a new plan A J B J C J D, we will reorder a, b, c again. Is my understanding right?